langchain/src/util/math_utils.ts

```diff
@@ -0,0 +1,134 @@
+import * as tf from "@tensorflow/tfjs-core";
+import { Tensor1D } from "@tensorflow/tfjs-core";
```

Comment:
This is a peer dep - would be nice to not have to use this.

Can we use this built-in instead?

https://github.com/hwchase17/langchainjs/blob/main/langchain/src/vectorstores/memory.ts#L1

And then calculate MMR off of that? It would make this more portable.

Reply:
Thanks for the review @jacoblee93! 
ml-distance and ml-matrix also work well for the cosine similarity calculation and tests.
Updated the code to use them instead.